unity/Assets/Scripts/Content/QuestData.cs

```diff
@@ -961,21 +965,25 @@ override public void ChangeReference(string oldName, string newName)
             // a next event is changed
             for (int i = 0; i < nextEvent.Count; i++)
             {
-                for (int j = 0; j < nextEvent[i].Count; j++)
+                List<string> eventNames = nextEvent[i].EventNames;
+                // Handle event removal
+                if (newName == string.Empty)
                 {
-                    if (nextEvent[i][j].Equals(oldName))
+                    eventNames.Remove(oldName);
+                    if (eventNames.Count <= 0 && !nextEvent[i].HasCondition)
                     {
-                        nextEvent[i][j] = newName;
+                        nextEvent.RemoveAt(i--);
                     }
+                    continue;
```

Comment:
In case an event is removed - should we remove the button if the condition is present?

Reply:
Yes I think this would be good to prevent unused objects.

Reply:
I just realized now that's not what the original implementation did. So I'll get rid of this entirely. Previously it was simply trimming the empty strings from the list of events.